silkworm/silkrpc/core/rawdb/chain.cpp

```diff
@@ -172,14 +172,13 @@ Task<Receipts> read_raw_receipts(const DatabaseReader& reader, BlockNum block_nu
     const auto block_key = silkworm::db::block_key(block_number);
     const auto data = co_await reader.get_one(db::table::kBlockReceiptsName, block_key);
     SILK_TRACE << "read_raw_receipts data: " << silkworm::to_hex(data);
+    Receipts receipts{};
     if (data.empty()) {
-        co_return Receipts{};  // TODO(canepat): use std::null_opt with Task<std::optional<Receipts>>?
+        co_return receipts;
```

Comment:
I think we need to change the return type of `read_raw_receipts` to be `Task<std::optional<Receipts>>` as per my comment and return `std::null_opt` here because this case is "there is no data for specified block in the db".

Then, we need to check the optional value at the call site in `get_receipts`.

All this thing is needed to distinguish among three cases:
- we have data in the db and it's valid
- we have no data in the db (i.e. pruning) and we may need to recompute the receipts by executing the tx
- we have an error decoding the data